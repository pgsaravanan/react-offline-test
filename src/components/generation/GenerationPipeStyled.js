import styled from 'styled-components';

export const GenerationStyled = styled.div`
    font-family: Arial;
`;

export const GenerationPotStyled = styled.ul`
    font-family: "Arial";
    font-size: 14px;
    list-style-type: none;
    margin: 0;
    padding: 0;
    width: 98%;
    display: flex;
    flex-wrap: wrap;
    li{
        span {
            color: black;
            text-transform: Capitalize;
        }
        padding: 5px 0;
        width: 20%;
        display: flex;
        .generation-fuel{
            display: inline-block;
            min-width: 80px;
        }
        .generation-pot{
            display: flex;
            border-radius: 50%;
            width: 100px;
            height: 100px;
            background-color: green;
            align-items: center;
            .nested {
                margin: 0 auto;
            }
        }
    }
`

export const GenerationPipeStyled = styled.ul`
    font-family: "Arial";
    font-size: 14px;
    list-style-type: none;
    margin: 0;
    padding: 0;
    width: 98%;
    
    li{
        span {
            color: black;
            text-transform: Capitalize;
        }
        padding: 5px 0;
        display: flex;
        .generation-fuel{
            display: inline-block;
            min-width: 80px;
        }
    }
`

const getBgColorForFuel = {
    biomass: '#9400D3',
    coal: '#4B0082',	
    imports:'#0000FF',
    gas:'#00FF00',
    nuclear:'#FFFF00',
    other:'#FF7F00',
    hydro:'pink',
    solar:'green',
    wind:'orange'
}

export const GenerationPipeItemStyled = styled.span`
    color: white;
    display: inline-block;
    background-color: red;
    transition: all ease 5s;
    width: 0%;
    padding: ${ ({perc}) => perc === 0 ? '0px': '5px'};
    width: ${ ({perc}) => `${perc}%`};
    border-radius: 4px;
    margin: 0 5px;
    &.hasRainbow{
        font-size: 100px;
        background-color: ${ ({fuel}) => {
            console.log(fuel);
            console.log(getBgColorForFuel[fuel])
            return fuel ? getBgColorForFuel[fuel] : 'red'
        }}
    }

`