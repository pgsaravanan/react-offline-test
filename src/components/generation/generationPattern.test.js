import React from 'react';
import {mount} from 'enzyme';
import GenerationPattern from './generationPattern';
import { GenerationPipeStyled, GenerationPipeItemStyled } from './GenerationPipeStyled';
import 'jest-styled-components';

describe('Generation Pattern', () => {

    it('should match snapshot', () => {
        const props = {
            generationmix: [{fuel: 'Fuel1', perc:1}],
            title: 'Test Title'
        }
        const tree =  mount(<GenerationPattern {...props} />)

        expect(tree).toMatchSnapshot();
    })

    it('should have title', () => {
        const props = {
            generationmix: [],
            title: 'Test Title'
        }
        const tree =  mount(<GenerationPattern {...props} />)

        expect(tree.find('h1').text()).toEqual('Test Title');
    })

    it('should have style className', () => {
        const props = {
            generationmix: [{fuel: 'Fuel1', perc:1}],
            title: 'Test Title',
            style: 'hasRainbow'
        }
        const tree =  mount(<GenerationPattern {...props} />)

        expect(tree.find(GenerationPipeItemStyled).hasClass('hasRainbow')).toBeTruthy();
    })

    it('should set pipe width from percentage attribute in generation data', () => {
        const props = {
            generationmix: [{fuel: 'Fuel1', perc:12.5}],
            title: 'Test Title',
            style: 'hasRainbow'
        }
        const tree =  mount(<GenerationPattern {...props} />)

        expect(tree.find(GenerationPipeItemStyled)).toHaveStyleRule('width', '12.5%');
    })

    describe('hasRainbow style', () => {

        it('should set pipe bg color for biomass', () => {
            const props = {
                generationmix: [{fuel: 'biomass', perc:12}],
                title: 'Test Title',
                style: 'hasRainbow'
            }
            const tree =  mount(<GenerationPattern {...props} />)
            expect(tree.find('.hasRainbow')).toHaveStyleRule('background-color', '#9400D3');
        })

        it('should set pipe bg color for coal', () => {
            const props = {
                generationmix: [{fuel: 'coal', perc:12}],
                title: 'Test Title',
                style: 'hasRainbow'
            }
            const tree =  mount(<GenerationPattern {...props} />)
            expect(tree.find('.hasRainbow')).toHaveStyleRule('background-color', '#4B0082');
        })

    })

    it('should set pipe bg color', () => {
        const props = {
            generationmix: [{fuel: 'Fuel1', perc:12.5}],
            title: 'Test Title',
        }
        const tree =  mount(<GenerationPattern {...props} />)

        expect(tree.find(GenerationPipeItemStyled)).toHaveStyleRule('background-color', 'red');
    })

    it('should have generation list item', () => {
        const props = {
            generationmix: [{fuel: 'Fuel1', perc:1},{fuel: 'Fuel2', perc:2}],
            title: 'Test Title'
        }
        const tree =  mount(<GenerationPattern {...props} />)

        expect(tree.find(GenerationPipeStyled).find('li').length).toEqual(2);
    })

})