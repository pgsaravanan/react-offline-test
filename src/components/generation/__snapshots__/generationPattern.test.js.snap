// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generation Pattern should match snapshot 1`] = `ReactWrapper {}`;
